backend/src/seed/factories/comments.js

```diff
@@ -1,21 +1,38 @@
 import faker from 'faker'
 import uuid from 'uuid/v4'
 
-export default function(params) {
-  const {
-    id = uuid(),
-    postId = 'p6',
-    content = [faker.lorem.sentence(), faker.lorem.sentence()].join('. '),
-  } = params
-
+export default function create() {
   return {
-    mutation: `
-      mutation($id: ID!, $postId: ID!, $content: String!) {
-        CreateComment(id: $id, postId: $postId, content: $content) {
-          id
-        }
+    factory: async ({ args, neodeInstance, factoryInstance }) => {
+      const defaults = {
+        id: uuid(),
+        content: [faker.lorem.sentence(), faker.lorem.sentence()].join('. '),
+      }
+      args = {
+        ...defaults,
+        ...args,
       }
-    `,
-    variables: { id, postId, content },
+      args.contentExcerpt = args.contentExcerpt || args.content
+
+      let { post, postId } = args
+      delete args.post
+      delete args.postId
+      if (post && postId) throw new Error('You provided both post and postId')
+      if (postId) post = await neodeInstance.find('Post', postId)
+      post = post || (await factoryInstance.create('Post'))
+
+      let { author, authorId } = args
+      delete args.author
+      delete args.authorId
+      if (author && authorId) throw new Error('You provided both author and authorId')
+      if (authorId) author = await neodeInstance.find('User', authorId)
+      author = author || (await factoryInstance.create('User'))
+
+      delete args.author
+      const comment = await neodeInstance.create('Comment', args)
+      await comment.relateTo(post, 'post')
+      await comment.relateTo(author, 'author')
+      return comment
+    },
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T11:48:15Z" title="Tuesday, September 3rd 2019, 1:48:15 pm +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,21 +1,38 @@
 import faker from 'faker'
 import uuid from 'uuid/v4'
 
-export default function(params) {
-  const {
-    id = uuid(),
-    postId = 'p6',
-    content = [faker.lorem.sentence(), faker.lorem.sentence()].join('. '),
-  } = params
-
+export default function create() {
   return {
-    mutation: `
-      mutation($id: ID!, $postId: ID!, $content: String!) {
-        CreateComment(id: $id, postId: $postId, content: $content) {
-          id
-        }
+    factory: async ({ args, neodeInstance, factoryInstance }) => {
+      const defaults = {
+        id: uuid(),
+        content: [faker.lorem.sentence(), faker.lorem.sentence()].join('. '),
+      }
+      args = {
+        ...defaults,
+        ...args,
       }
-    `,
-    variables: { id, postId, content },
+      args.contentExcerpt = args.contentExcerpt || args.content
+
+      let { post, postId } = args
+      delete args.post
+      delete args.postId
+      if (post && post) throw new Error('You provided both post and postId')
+      if (postId) post = await neodeInstance.find('Post', postId)
+      post = post || (await factoryInstance.create('Post'))
+
+      let { author, authorId } = args
+      delete args.author
+      delete args.authorId
+      if (author && authorId) throw new Error('You provided both author and authorId')
+      if (authorId) author = await neodeInstance.find('User', authorId)
+      author = author || (await factoryInstance.create('User'))
+
+      delete args.author
+      const comment = await neodeInstance.create('Comment', args)
+      await comment.relateTo(post, 'post')
+      await comment.relateTo(author, 'author')
+      return comment
+    },
```


you refactored the factories!!!??
you rock!!!